packages/edit-site/src/components/layout/router.js

```diff
@@ -35,6 +37,7 @@ export default function useLayoutAreas() {
 		return {
 			key: 'pages-list',
 			areas: {
+				sidebar: <Sidebar router={ router } />,
```

Comment:
It's unfortunate that we keep rendering a generic "Sidebar" component here. I wanted the routes to be more clear here. (Basically move the sidebar routing into this component instead) 

Today it feels like we're defining routes twice.

Reply:
Good point. 🙂  I'll try to deconstruct `Sidebar` into individual items. That should make the `Navigator` stuff disappear sooner or later.

Reply:
In 06000f3 I made some progress on this. The `Sidebar` component is completely dissolved and all the individual items are put into `area.sidebar`.

There is still `NavigationProvider`, but it's there to:
- expose APIs used by `useNavigator()`, like the `goTo` or `goToParent` methods, and the current `location`.
- keep the "navigator" styles on the rendered elements.

I no longer need the code that registers screens in the router: there's now one hardcoded `SCREENS` array used to map paths and determine parent paths.

I'll continue working on eliminating these remaining parts of the `Navigator` router, too.

Reply:
This already feels like a great improvement and a step in the right direction 👍

Reply:
The use of `Navigator` components is now almost completely eliminated, the remaining usages are now mere shadows 🙂 

The items in the Site Editor sidebar are still `NavigatorButton`s:

<img width="357" alt="Screenshot 2024-04-15 at 9 19 41" src="https://github.com/WordPress/gutenberg/assets/664258/f6c2514d-71ab-4066-a590-6e9d18d4de44">

They use the `goTo` method of `useNavigator()`. I'm going to remove that now, and replace `goTo` with `history.push`.

Each of the sidebar screens is also still wrapped in `NavigatorScreen`. It reads the `location` from `useNavigator`, but the only properties it looks at are things related to focus management. It _doesn't_ look at the _actual_ location at all. Focus management is the last big thing to solve in this PR, I was postponing it until now.

I'm adding @oandregal as a reviewer for awareness, because I'm noticing the work being done on eliminating `/wp_template_part/all` and sometimes there are merge conflicts.

Reply:
Today I eliminated even the last usages of `Navigator`. It's not there at all in Site Editor sidebar and routing.

Two things I need to reconstruct before this is ready for merge:
1. Animations when navigating forward and back in the Sidebar. When pressing the Back button, we want the next screen to slide from the left. When pressing one of the "forward" buttons, i.e., the one of the items with chevrons, we want to slide from the right.
2. Focus management. When navigating forward, focus the "back" button. When navigating backward, focus the item leading to where we came from. Pressing Enter repeatedly should navigate back and forth between the same two items.

I think both of these behaviors are local to the Sidebar, and could be handled in its local state. The global router doesn't necessarily need to know about it.